src/basic/recurse-dir.c

```diff
@@ -101,11 +110,14 @@ int readdir_all(int dir_fd,
 
         j = 0;
         FOREACH_DIRENT_IN_BUFFER(entry, de->buffer, de->buffer_size) {
-                if (ignore_dirent(entry, flags))
+                /* If d_type == DT_UNKNOWN that means we failed to ensure the type in the earlier loop and
+                 * didn't include the dentry in de->n_entries and as such should skip it here as well. */
+                if (ignore_dirent(entry, flags) || entry->d_type == DT_UNKNOWN)
```

Comment:
you need to conditionalize on RECURSE_DIR_ENSURE_TYPE here, because otherwise DT_UNKNOWN is totally fine.

Reply:
Ughhh I don't know what's up with me today

Reply:
I need more coffee